src/ol/tilegrid/TileGrid.js

```diff
@@ -262,8 +265,8 @@ class TileGrid {
     }
     while (z >= this.minZoom) {
       if (this.zoomFactor_ === 2) {
-        x = Math.floor(x / 2);
-        y = Math.floor(y / 2);
+        x = Math.floor(/** @type {number} */ (x) / 2);
+        y = Math.floor(/** @type {number} */ (y) / 2);
```

Comment:
I'm wondering if these type casts could be avoided when the wrapping if clause is changed from
```js
if (this.zoomFactor_ === 2) {
```
to
```js
if (x !== undefined && y !== undefined) {
```

Reply:
That works.